src/calva-fmt/src/infer.ts

```diff
@@ -97,15 +98,34 @@ function applyResults(r: ResultOptions, editor: vscode.TextEditor) {
     void editor
       .edit(
         (editBuilder) => {
+          if (isUndefined(r.edits)) {
+            console.error('Edits were undefined!', cloneDeep({ editBuilder, r, editor }));
+            return;
+          }
           r.edits.forEach((edit: CFEdit) => {
             const start = new vscode.Position(edit.start.line, edit.start.character),
               end = new vscode.Position(edit.end.line, edit.end.character);
+            if (isUndefined(edit.text)) {
+              console.error(
+                'edit.text was undefined!',
+                cloneDeep({ edit, editBuilder, r, editor })
+              );
+              return;
+            }
             editBuilder.replace(new vscode.Range(start, end), edit.text);
           });
         },
         { undoStopAfter: true, undoStopBefore: false }
       )
       .then((_onFulfilled: boolean) => {
+        if (isUndefined(r.line)) {
+          console.error('Expected result line to be defined!', { r });
+          return;
+        }
+        if (isUndefined(r.character)) {
+          console.error('Expected result character to be defined!', { r });
+          return;
+        }
```

Comment:
I don't know if, in practice, these will ever actually be undefined or not. It would be nice to have the types reflect that if that's the case.

Reply:
They will never be undefined. This is the code we use to calculate them:

```clojure
(defn raplacement-edits-for-diffing-lines
  "Returns a list of replacement edits to apply to `old-text` to get `new-text`.
   Edits will be in the form `[:replace [range] text]`,
   where `range` is in the form `[[start-line start-char] [end-line end-char]]`.
   NB: The two versions need to have the same amount of lines."
  [old-text new-text]
  (let [old-lines (util/split-into-lines old-text)
        new-lines (util/split-into-lines new-text)]
    (->> (map vector (range) old-lines new-lines)
         (remove (fn [[line o n]] (= o n)))
         (mapv (fn [[line o n]]
                 {:edit "replace"
                  :start {:line line
                          :character 0}
                  :end {:line line
                        :character (count o)}
                  :text n})))))
```

Reply:
cool, I'll modify these to use non-null-assertions (`!`) then